lib/pageflow/zencoder_api.rb

```diff
@@ -36,19 +36,35 @@ def get_info(job_id)
       end
     end
 
+    # @deprecated
     def get_input_details(job_id)
+      get_details(job_id)
+    end
+
+    def get_details(job_id)
       with_exception_translation do
         response = Zencoder::Job.details(job_id)
 
         if response.success?
           input_details = response.body['job']['input_media_file']
+          outputs_details = response.body['job']['output_media_files']
 
-          {
-            :format => input_details["format"],
-            :duration_in_ms => input_details["duration_in_ms"],
-            :width => input_details["width"],
-            :height => input_details["height"]
+          returned_details = {
+            format: input_details['format'],
+            duration_in_ms: input_details['duration_in_ms'],
+            width: input_details['width'],
+            height: input_details['height'],
+            outputs_presences: {}
           }
+
+          outputs_details.each do |output|
+            if output['label'].present?
+              returned_details[:outputs_presences]
+                .merge!(output['label'].to_sym => output['state'])
```

Comment:
This can be expressed in a more functional style with `reduce` instead of using destructive merge